aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2532,8 +2674,10 @@ Tensor transpose(const Tensor & self, int64_t dim0, int64_t dim1) {
     Tensor self_clone = self.clone();
     return sparse_transpose_(self_clone, dim0, dim1);
   }
-  TORCH_CHECK(!(self.layout() == kSparseBsr || self.layout() == kSparseBsc),
-      "Transposition of tensors with ", self.layout(), " layout is currently not supported.");
+  if (self.layout() == kSparseBsr || self.layout() == kSparseCsr ||
+      self.layout() == kSparseBsc || self.layout() == kSparseCsc) {
```

Comment:
do we have a util for that? https://github.com/pytorch/pytorch/pull/82167/ could be of help.

Reply:
Think that util is intended to be used for this type of expression exactly. Should one of us take over that PR?

Reply:
@nikitaved @amjames not a strong opinion, but personally I'd rather see a small handful of handwritten utility functions - `has_sparse_csr_layout()`, etc.

I like that better because the groupings being tested are semantically meaningful - so like, we add a new sparse layout and we can just go to this handful of util functions and it's obvious which ones to update and we're done, rather than having to grep a bunch of call sites and figure out intent.

The other more pedantic reason to prefer handwritten functions is that the template func in https://github.com/pytorch/pytorch/pull/82167 is vulnerable to the combinatorial codegen explosion thing - it'll generate a new function for every unique sequence of layout arguments. Total biomass is not gonna be that big ofc but in general I definitely think templates should be avoided except when the ROI justifies the increase in code obscurity/opacity 😁 

Case in point: 82167 doesn't eliminate the multiple calls to `self.layout()`, but that's much harder to see in `foo(self.layout(), x, y, z)` than in `self.layout() == x || self.layout() == y || self.layout() == z`